lib/dependencies/CommonJsImportsParserPlugin.js

```diff
@@ -375,14 +450,192 @@ class CommonJsImportsParserPlugin {
 
 		parser.hooks.call
 			.for("require.resolve")
-			.tap("RequireResolveDependencyParserPlugin", expr => {
+			.tap("CommonJsImportsParserPlugin", expr => {
 				return processResolve(expr, false);
 			});
 		parser.hooks.call
 			.for("require.resolveWeak")
-			.tap("RequireResolveDependencyParserPlugin", expr => {
+			.tap("CommonJsImportsParserPlugin", expr => {
 				return processResolve(expr, true);
 			});
+		//#endregion
+
+		//#region Create require
+		/**
+		 * @param {CallExpressionNode} expr call expression
+		 * @returns {string|boolean} context
+		 */
+		const parseCreateRequireArguments = expr => {
+			const args = expr.arguments;
+			if (args.length !== 1) {
+				const err = new WebpackError(
+					"module.createRequire supports only one argument."
+				);
+				err.loc = expr.loc;
+				parser.state.module.addWarning(err);
+				return;
+			}
+			const createParsingError = node => {
+				const err = new WebpackError(
+					"module.createRequire() failed parsing argument."
+				);
+				err.loc = node.loc;
+				parser.state.module.addWarning(err);
+			};
+			const arg = args[0];
+			if (
+				arg.type === "MemberExpression" &&
+				arg.object.type === "MetaProperty"
+			) {
+				if (
+					arg.object.meta.type === "Identifier" &&
+					arg.object.meta.name === "import" &&
+					arg.object.property.type === "Identifier" &&
+					arg.object.property.name === "meta" &&
+					arg.property.type === "Identifier" &&
+					arg.property.name === "url"
+				) {
+					// same module context
+					return false;
```

Comment:
I think we evaluate `import.meta` to an identifier in `evaluateExpression`, so this special case isn't needed in my opinion.

You can use `evaluated.isIdentifier() && evaluated.identifier === "import.meta.url"` for that case in the code below